### DISCUSSION TITLE
How to use the gluestack-ui types

### DISCUSSION THREAD

**Discussion by samislam** [gluestack/gluestack-ui#1182] [Q&A]
I am using the new components library called ([gluestack-ui][1]).

I am using typescript and I want to use the "Box" component props types interface in my component.

I can't find, is the interface exported?

```tsx
import { Box } from '@gluestack-ui/themed'

export interface PaperProps extends PropsWithChildren  {}

export default function Paper({ children }: PaperProps) { 
  return <Box>{children}</Box>
}

```

In  material-ui for example, you can:

```
import { Box, type BoxProps } from '@mui/material'
```

the  I can for example:
```
export interface PaperProps extends PropsWithChildren & BoxProps  {}
```

How can I do the same with gluestack-ui?

  [1]: https://ui.gluestack.io

**Comment by ts-candide** [comment]
I struggled with this too until I discovered `ComponentProps`:

``` ts
import { Box } from "@gluestack-ui/themed"
import React, { ComponentProps } from "react"

type BoxProps = ComponentProps<typeof Box>
```

**Reply by mattaningram** [reply]
I also use this and it works fine mostly except for some reason if I am extending types that have a `shift` prop then extending with Box props doesn't work. It complains that the `shift` property isn't compatible, but as far as I can tell the `Box` element doesn't have a `shift` property! Have no idea how to go about debugging/fixing this.
